lib/ui/profile/SwitchProfile.dart

```diff
@@ -0,0 +1,208 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildSimpleAppBar(context),
+      body: _buildBody(),
+      floatingActionButton: _buildFloatingButton(),
+      floatingActionButtonLocation: FloatingActionButtonLocation.centerDocked,
+    );
+  }
+
+  _buildBody() {
+    return SingleChildScrollView(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Padding(
+            padding: _Constants.generalEdgePadding,
+            child: Text(
+              widget._viewModel.title,
+              textAlign: TextAlign.left,
+              style: widget._style.titleTextStyle,
+            ),
+          ),
+          ListView.separated(
+            shrinkWrap: true,
+            physics: ScrollPhysics(),
+            itemBuilder: (context, index) => SwitchProfileItems(
+              viewModel: SwitchProfileItemsViewModel(
+                title: widget._viewModel.actions[index].title,
+                image: widget._viewModel.actions[index].image,
+                icon: widget._viewModel.actions[index].icon,
+                isSelected: widget._viewModel.actions[index].isSelected,
+                itemAction: () => _select(index),
+              ),
+            ),
+            separatorBuilder: (context, index) => ListDivider.build(),
+            itemCount: widget._viewModel.actions.length,
+          ),
+        ],
+      ),
+    );
+  }
+
+  _buildFloatingButton() {
+    return Padding(
+      padding: _Constants.bottomButtonEdgePadding,
+      child: Visibility(
+        visible: widget._viewModel.isVisible,
+        child: RoundedButton(
+            viewModel: RoundedButtonViewModel(
+                title: widget._viewModel.title,
+                onTap: () => _switchProfileTapped()),
+            style: RoundedButtonStyle.largeRoundedButtonStyle()),
+      ),
+    );
+  }
+
+  AppBar _buildSimpleAppBar(BuildContext context) {
+    return AppBar(
+      elevation: _Constants.appBarElevation,
+      leading: FlatButton(
+        onPressed: () => Navigator.pop(context, false),
+        padding: _Constants.appBarEdgePadding,
+        child: Image.asset(
+          _Icons.dismissModal,
+          height: _Constants.appBarIconSize,
+          width: _Constants.appBarIconSize,
+        ),
+      ),
+      actions: <Widget>[
+        Center(
+          child: Padding(
+            padding: _Constants.topButtonEdgePadding,
+            child: RoundedButton(
+                viewModel: RoundedButtonViewModel(
+                    icon: _Icons.topButton,
+                    onTap: () => widget._viewModel.addProfileAction()),
+                style: RoundedButtonStyle.defaultStyle()),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _select(int index) {
+    setState(() {
+      widget._viewModel.selectedIndex = index;
+      _clearSelection();
+      widget._viewModel.actions[index].isSelected = true;
```

Comment:
the view shouldn't directly mutate the view model like this... this should be delegated to the viewmodel. viewModel.selectItem(itemViewmodel)